doc/trajopt_planner/src/trajopt_example.cpp

```diff
@@ -123,6 +129,49 @@ int main(int argc, char** argv)
   const Eigen::Isometry3d& end_effector_transform_goal = robot_state->getGlobalLinkTransform(link_model_names.back());
   pose_msg_goal = tf2::toMsg(end_effector_transform_goal);
 
+  // Reference Trajectory. The type should be defined in the yaml file.
+  // ========================================================================================
+  // type: STATIONARY
+  // No need to pass any trajectory. The current joint values will be replicated for all timesteps
+
+  // type: JOINT_INTERPOLATED
+  // The joint values at a specified state. Could be the goal state or one of the goals when having multiple goal states
+  // The first index (points[0]) is used to set the values of the joints at the specified state as follows:
+  // req.reference_trajectories.resize(1);
+  // req.reference_trajectories[0].joint_trajectory.resize(1);
+  // req.reference_trajectories[0].joint_trajectory[0].joint_names = joint_names;
+  // req.reference_trajectories[0].joint_trajectory[0].points.resize(1);
+  // req.reference_trajectories[0].joint_trajectory[0].points[0].positions = goal_joint_values;
+
+  // type: GIVEN_TRAJ
+  // For this example, we give an interpolated trajectory
+  int n_steps = 20; // number of steps
+  int n_dof = 7; // number of degrees of freedom
+
+  std::vector<double> dt_vector;
+  for (int joint_index = 0; joint_index < n_dof; ++joint_index)
+  {
+    double dt = (goal_joint_values[joint_index] - current_joint_values[joint_index]) / n_steps;
+    dt_vector.push_back(dt);
+  }
+
+  req.reference_trajectories.resize(1);
+  req.reference_trajectories[0].joint_trajectory.resize(1);
+  req.reference_trajectories[0].joint_trajectory[0].points.resize(n_steps + 1); // trajectory includes both the start and end points (n_steps + 1)
+  req.reference_trajectories[0].joint_trajectory[0].joint_names = joint_names;
+  req.reference_trajectories[0].joint_trajectory[0].points[0].positions = current_joint_values;
+  for (std::size_t stp_index = 1; stp_index <= n_steps; ++stp_index)
+  {
+    std::vector<double> j_values;
+    for (int dof_index = 0; dof_index < n_dof; ++dof_index)
+    {
+      double j_value = current_joint_values[dof_index] + stp_index * dt_vector[dof_index];
```

Comment:
Please write it out, `joint_value`.in the above part you didn't abbreviate `joint_index` either which I think for a purpose of tutorial code is very beneficial.